### ISSUE TITLE
data_FACED reshape problem

### ISSUE THREAD

**Issue by Student-xtLin** [Miracle-2001/GNN4EEG#8]
in your code `protocol.py `，the method `data_FACED( )`, line 144, `data = data.reshape(-1, feature_shape, 30).transpose([0, 2, 1])`. Are u sure your reshape is correct?  or  `data = data.reshape(-1,  30, feature_shape)` is correct？

```
def data_FACED(protocol, categories, data_path):
    if protocol != "cross_subject" and protocol != "intra_subject":
        raise ValueError(
            "The protocol should be either 'cross_subject' or 'intra_subject'.")
    if categories != 2 and categories != 9:
        raise ValueError("The categories should be either 2 or 9.")
    if os.path.exists(data_path) == False:
        raise ValueError("The path of FACED dataset does not exist.")

    if categories != 2 and categories != 9:
        raise ValueError(
            "The label categories in FACED dataset should be either 2 or 9.")

    # Note: the signals are in dict['de_lds']
    data = hdf5.loadmat(data_path)['de_lds']
    # data shape: (123, 720 or 840, 120)  120=4*30  4 band and 30 nodes
    label_type = 'cls2' if categories == 2 else 'cls9'
    data, label_repeat, n_samples = load_srt_de(
        data, True, False, 1, label_type)
    # label_repeat shape: 720 or 840   720=24*30. 840=28*30   24/28 videos and 30 samples are generated by each video

    feature_shape = int(data.shape[-1]/30)
    labels = np.tile(label_repeat, data.shape[0])
    data = data.reshape(-1, feature_shape, 30).transpose([0, 2, 1])

    subject_id_list = [int(i/len(label_repeat))
                       for i in range(labels.shape[0])]
    dataloader = DataLoader(protocol, data, labels,
                            'FACED', subject_id_list=subject_id_list)
    return dataloader

```

**Comment by Miracle-2001** [comment]
Good question. That's because the data with shape&nbsp;(123, 720 or 840, 120), and the "120" is reshaped from (4,30) to 120 before ( not (30,4) to 120 ; and you can find that code on the official FACED dataset website). 
So here when one need to split the "120"， you have to first reshape to (4,30) .&nbsp;
That's why it is&nbsp;data = data.reshape(-1, feature_shape, 30).transpose([0, 2, 1])&nbsp;rather than&nbsp;&nbsp;data = data.reshape(-1, 30, feature_shape)&nbsp;
Thanks for your attention, hope things work out~
&nbsp;


*Miracle*
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月15日(星期天) 下午3:42
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; 
主题: [Miracle-2001/GNN4EEG] data_FACED reshape problem (Issue #8)














 
in your code protocol.py ，the method data_FACED( ), line 144, data = data.reshape(-1, feature_shape, 30).transpose([0, 2, 1]). Are u sure your reshape is correct?  or  data = data.reshape(-1,  30, feature_shape) is correct？
 def data_FACED(protocol, categories, data_path):     if protocol != "cross_subject" and protocol != "intra_subject":         raise ValueError(             "The protocol should be either 'cross_subject' or 'intra_subject'.")     if categories != 2 and categories != 9:         raise ValueError("The categories should be either 2 or 9.")     if os.path.exists(data_path) == False:         raise ValueError("The path of FACED dataset does not exist.")     if categories != 2 and categories != 9:         raise ValueError(             "The label categories in FACED dataset should be either 2 or 9.")     # Note: the signals are in dict['de_lds']     data = hdf5.loadmat(data_path)['de_lds']     # data shape: (123, 720 or 840, 120)  120=4*30  4 band and 30 nodes     label_type = 'cls2' if categories == 2 else 'cls9'     data, label_repeat, n_samples = load_srt_de(         data, True, False, 1, label_type)     # label_repeat shape: 720 or 840   720=24*30. 840=28*30   24/28 videos and 30 samples are generated by each video     feature_shape = int(data.shape[-1]/30)     labels = np.tile(label_repeat, data.shape[0])     data = data.reshape(-1, feature_shape, 30).transpose([0, 2, 1])     subject_id_list = [int(i/len(label_repeat))                        for i in range(labels.shape[0])]     dataloader = DataLoader(protocol, data, labels,                             'FACED', subject_id_list=subject_id_list)     return dataloader

**Comment by Student-xtLin** [comment]
get it， thx a lot.
